lib/mbr.mli

```diff
@@ -99,3 +99,9 @@ val sizeof : int
 
 val default_partition_start : int32
 (** default sector offset for first partition *)
+
+val print_mbr_fields : t -> unit
+(** print the contents of an MBR header *)
+
+val read_mbr : string -> t
+(** read an MBR and parse it's header **)
```

Comment:
I understand your motivation for putting `print_mbr_fields` and `read_mbr` and I would agree it would be nice to reuse the code. In the case of `read_mbr` however it depends on a Unix-like filesystem - something we generally don't have in Mirage OS. At best this function would not be usable in Mirage and at worst it would mean the unikernel wouldn't compile. As for `print_mbr_fields` I think it is fine, but users of the library may want to print the data differently. For this reason I think it is better left out, but I don't mind it being included.

Reply:
Oh it makes more sense now. I didn't initially understand the scope of how the functions will be used. I think both functions are better off in the `mbr_inspect.ml` file.